sabnzbd/newsunpack.py

```diff
@@ -2077,6 +2077,23 @@ def quick_check_set(set, nzo):
             if nzf.md5sum == md5pack[file]:
                 try:
                     logging.debug("Quick-check will rename %s to %s", nzf.filename, file)
+
+                    # Let's check if there is a subdir in 'file', which SABs needs to create
+                    relative_new_path = re.search(r"(.*)[\\|/]", file)  # find directory before separator \ or /
+                    # Proceed if subdir specified, and no malicous '..' in it
+                    if relative_new_path and not os.pardir in relative_new_path.group(1):
```

Comment:
I don't think this is sufficient because .* will match everything up to the file. `../../file` would result in `../..`